code/mapping/src/mapping_data_integration.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+
+from ros_compatibility.node import CompatibleNode
+import ros_compatibility as roscomp
+import ros_numpy
+import rospy
+
+import random
+
+from typing import List, Optional
+
+from mapping_common.entity import Entity, Flags
+from mapping_common.transform import Transform2D, Vector2
+from mapping_common.shape import Circle
+from mapping_common.map import Map
+from mapping.msg import Map as MapMsg
+
+from sensor_msgs.msg import PointCloud2
+
+
+class MappingDataIntegrationNode(CompatibleNode):
+    """Creates the initial map data frame based on all kinds of sensor data
+
+    Sends this map off to Filtering and other consumers (planning, acting)
+
+    This node sends the maps off at a fixed rate.
+    (-> It buffers incoming sensor data slightly)
+    """
+
+    lidar_data: Optional[PointCloud2] = None
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+        self.new_subscription(
+            topic=self.get_param("~lidar_topic", "/carla/hero/LIDAR"),
+            msg_type=PointCloud2,
+            callback=self.lidar_callback,
+            qos_profile=1,
+        )
+        self.map_publisher = self.new_publisher(
+            msg_type=MapMsg,
+            topic=self.get_param("~map_init_topic", "/paf/hero/mapping/init_data"),
+            qos_profile=1,
+        )
+        self.rate = 1.0 / 20.0
```

Comment:
Rate should be configurable in as a param. Add the param to the launch file as well